src/providers/BasicPoolsProvider.tsx

```diff
@@ -176,75 +186,43 @@ export async function getPoolsDataFromRegistry(
       registryPoolDataMultiCalls,
     )
 
-    const arePoolsPausedMulticalls: ReturnType<
-      typeof poolRegistryMultiCall.getPaused
-    >[] = []
-    const virtualPricesMulticalls: ReturnType<
-      typeof poolRegistryMultiCall.getVirtualPrice
-    >[] = []
-    const swapStoragesMulticalls: ReturnType<
-      typeof poolRegistryMultiCall.getSwapStorage
-    >[] = []
-    const aParametersMultiCalls: ReturnType<
-      typeof poolRegistryMultiCall.getA
-    >[] = []
-    const tokenBalancesMultiCalls: ReturnType<
-      typeof poolRegistryMultiCall.getTokenBalances
-    >[] = []
-    const underlyingTokenBalancesMultiCalls: ReturnType<
-      typeof poolRegistryMultiCall.getUnderlyingTokenBalances
-    >[] = []
-    const lpTokenTotalSuppliesMultiCalls: MulticallCall<unknown, BigNumber>[] =
-      []
-
-    registryPoolData.forEach((poolData) => {
-      if (poolData != null) {
-        const { lpToken, poolAddress } = poolData
-        const lpTokenContract = createMultiCallContract<Erc20>(
-          lpToken.toLowerCase(),
-          ERC20_ABI,
-        )
-
-        lpTokenTotalSuppliesMultiCalls.push(lpTokenContract.totalSupply())
-        arePoolsPausedMulticalls.push(
-          poolRegistryMultiCall.getPaused(poolAddress),
-        )
-        virtualPricesMulticalls.push(
-          poolRegistryMultiCall.getVirtualPrice(poolAddress),
-        )
-        swapStoragesMulticalls.push(
-          poolRegistryMultiCall.getSwapStorage(poolAddress),
-        )
-        aParametersMultiCalls.push(poolRegistryMultiCall.getA(poolAddress))
-        tokenBalancesMultiCalls.push(
-          poolRegistryMultiCall.getTokenBalances(poolAddress),
-        )
-        underlyingTokenBalancesMultiCalls.push(
-          poolRegistryMultiCall.getUnderlyingTokenBalances(poolAddress),
-        )
-      }
+    const poolMulticalls = (
+      registryPoolData.filter(Boolean) as {
+        lpToken: string
+        poolAddress: string
+      }[]
+    ).map(({ lpToken, poolAddress }) => {
+      const lpTokenContract = createMultiCallContract<Erc20>(
+        lpToken.toLowerCase(),
+        ERC20_ABI,
+      )
+      return [
+        lpTokenContract.totalSupply(),
+        poolRegistryMultiCall.getA(poolAddress),
+        poolRegistryMultiCall.getPaused(poolAddress),
+        poolRegistryMultiCall.getSwapStorage(poolAddress),
+        poolRegistryMultiCall.getTokenBalances(poolAddress),
+        poolRegistryMultiCall.getUnderlyingTokenBalances(poolAddress),
+        poolRegistryMultiCall.getVirtualPrice(poolAddress),
+      ] as const
```

Comment:
using `as const` allows us to keep the exact types of each item in the tuple (no longer seen as an array)

Reply:
also sorted by name for easier scanning